clarifai/client/input.py

```diff
@@ -0,0 +1,507 @@
+        return self.upload_inputs([input_pb])
+
+    def upload_images_from_folder(self, folder_path: str, dataset_id: str = None) -> str:
+        """Upload images from folder.
+        The folder should only contain images. The filename of the image is used as the input_id.
+        Args:
+            folder_path (str): Path to the folder containing images.
+        Returns:
+            input_job_id (str): job id for the upload request.
+        Example:
+            >>> from clarifai.client.input import Input
+            >>> input_obj = Input(user_id = 'user_id', app_id = 'demo_app')
+            >>> input_obj.upload_from_folder(folder_path='demo_folder')
+        """
+        inputs = self.get_image_inputs_from_folder(folder_path, dataset_id)
+        return self._bulk_upload(inputs)
+
+    def upload_inputs(self, inputs: List[Input], show_log: bool = True) -> str:
+        """Upload list of input objects to the app.
+        Args:
+            inputs (list): List of input objects to upload.
+            show_log (bool): Show upload status log.
+        Returns:
+            input_job_id: job id for the upload request.
+        """
+        if not isinstance(inputs, list):
+            raise Exception("inputs must be a list of Input objects")
+        input_job_id = uuid.uuid4().hex  # generate a unique id for this job
+        request = service_pb2.PostInputsRequest(
+            user_app_id=self.user_app_id, inputs=inputs, inputs_add_job_id=input_job_id)
+        response = self._grpc_request(self.STUB.PostInputs, request)
+        if response.status.code != status_code_pb2.SUCCESS:
+            try:
+                self.logger.warning(response.inputs[0].status)
+            except IndexError:
+                self.logger.warning(response.status)
+        else:
+            if show_log:
+                self.logger.info("\nInputs Uploaded\n%s", response.status)
+
+        return input_job_id
+
+    def _upload_batch(self, inputs: List[Input]) -> List[Input]:
+        """Upload a batch of input objects to the app.
+        Args:
+            inputs (List[Input]): List of input objects to upload.
+        Returns:
+            input_job_id: job id for the upload request.
+        """
+        input_job_id = self.upload_inputs(inputs, False)
+        self._wait_for_inputs(input_job_id)
+        failed_inputs = self._delete_failed_inputs(inputs)
+
+        return failed_inputs
+
+    def delete_inputs(self, inputs: List[Input]) -> None:
+        """Delete list of input objects from the app.
+        Args:
+            input_ids (Input): List of input objects to delete.
+        Example:
+            >>> from clarifai.client.user import User
+            >>> input_obj = User(user_id="user_id").app(app_id="app_id").inputs()
+            >>> input_obj.delete_inputs(input_obj.list_inputs())
+        """
+        if not isinstance(inputs, list):
+            raise Exception("input_ids must be a list of input ids")
```

Comment:
Could we change all `ValueErrors` and relevant `Exception` to `UserError`, so that it's consistent through out the SDK